src/plugins/discover/public/application/components/discover_grid/discover_grid_document_selection.test.tsx

```diff
@@ -0,0 +1,143 @@
+      const contextMock = {
+        expanded: undefined,
+        setExpanded: jest.fn(),
+        rows: esHits,
+        onFilter: jest.fn(),
+        indexPattern: indexPatternMock,
+        isDarkMode: false,
+        selectedDocs: [],
+        setSelectedDocs: jest.fn(),
+      };
+
+      const component = mountWithIntl(
+        <DiscoverGridContext.Provider value={contextMock}>
+          <SelectButton rowIndex={0} />
+        </DiscoverGridContext.Provider>
+      );
+
+      const checkBox = findTestSubject(component, 'dscGridSelectDoc-i::1::');
+      expect(checkBox.props().checked).toBeFalsy();
+    });
+
+    test('is checked', () => {
+      const contextMock = {
+        expanded: undefined,
+        setExpanded: jest.fn(),
+        rows: esHits,
+        onFilter: jest.fn(),
+        indexPattern: indexPatternMock,
+        isDarkMode: false,
+        selectedDocs: ['i::1::'],
+        setSelectedDocs: jest.fn(),
+      };
+
+      const component = mountWithIntl(
+        <DiscoverGridContext.Provider value={contextMock}>
+          <SelectButton rowIndex={0} />
+        </DiscoverGridContext.Provider>
+      );
+
+      const checkBox = findTestSubject(component, 'dscGridSelectDoc-i::1::');
+      expect(checkBox.props().checked).toBeTruthy();
+    });
+
+    test('adding a selection', () => {
+      const contextMock = {
+        expanded: undefined,
+        setExpanded: jest.fn(),
+        rows: esHits,
+        onFilter: jest.fn(),
+        indexPattern: indexPatternMock,
+        isDarkMode: false,
+        selectedDocs: [],
+        setSelectedDocs: jest.fn(),
+      };
+
+      const component = mountWithIntl(
+        <DiscoverGridContext.Provider value={contextMock}>
+          <SelectButton rowIndex={0} />
+        </DiscoverGridContext.Provider>
+      );
+
+      const checkBox = findTestSubject(component, 'dscGridSelectDoc-i::1::');
+      checkBox.simulate('change');
+      expect(contextMock.setSelectedDocs).toHaveBeenCalledWith(['i::1::']);
+    });
+    test('removing a selection', () => {
+      const contextMock = {
+        expanded: undefined,
+        setExpanded: jest.fn(),
+        rows: esHits,
+        onFilter: jest.fn(),
+        indexPattern: indexPatternMock,
+        isDarkMode: false,
+        selectedDocs: ['i::1::'],
+        setSelectedDocs: jest.fn(),
+      };
+
+      const component = mountWithIntl(
+        <DiscoverGridContext.Provider value={contextMock}>
+          <SelectButton rowIndex={0} />
+        </DiscoverGridContext.Provider>
+      );
+
+      const checkBox = findTestSubject(component, 'dscGridSelectDoc-i::1::');
+      checkBox.simulate('change');
+      expect(contextMock.setSelectedDocs).toHaveBeenCalledWith([]);
+    });
+  });
+  describe('DiscoverGridDocumentToolbarBtn', () => {
+    test('it renders a button clickable button', () => {
+      const props = {
+        isFilterActive: false,
+        rows: esHits,
+        selectedDocs: ['i::1::'],
```

Comment:
what is this `::` notation?

Reply:
Had to build an `id` for the selected document, since _id is not unique, it's `{_index}::{_id}::{_routing}`